src/clp_ffi_js/ir/UnstructuredIrStreamReader.cpp

```diff
@@ -155,38 +132,9 @@ auto UnstructuredIrStreamReader::deserialize_stream() -> size_t {
 
 auto UnstructuredIrStreamReader::decode_range(size_t begin_idx, size_t end_idx, bool use_filter)
         const -> DecodedResultsTsType {
-    if (use_filter && false == m_filtered_log_event_map.has_value()) {
-        return DecodedResultsTsType{emscripten::val::null()};
-    }
-
-    size_t length{0};
-    if (use_filter) {
-        length = m_filtered_log_event_map->size();
-    } else {
-        length = m_encoded_log_events.size();
-    }
-    if (length < end_idx || begin_idx > end_idx) {
-        return DecodedResultsTsType{emscripten::val::null()};
-    }
-
-    std::string message;
-    constexpr size_t cDefaultReservedMessageLength{512};
-    message.reserve(cDefaultReservedMessageLength);
-    auto const results{emscripten::val::array()};
-
-    for (size_t i = begin_idx; i < end_idx; ++i) {
-        size_t log_event_idx{0};
-        if (use_filter) {
-            log_event_idx = m_filtered_log_event_map->at(i);
-        } else {
-            log_event_idx = i;
-        }
-        auto const& log_event_with_filter_data{m_encoded_log_events[log_event_idx]};
-        auto const& unstructured_log_event = log_event_with_filter_data.get_log_event();
-        auto const& log_level = log_event_with_filter_data.get_log_level();
-        auto const& timestamp = log_event_with_filter_data.get_timestamp();
-
-        auto const parsed{unstructured_log_event.get_message().decode_and_unparse()};
+    auto log_event_to_string = [this](UnstructuredLogEvent const& log_event) -> std::string {
+        std::string message;
+        auto const parsed{log_event.get_message().decode_and_unparse()};
         if (false == parsed.has_value()) {
             throw ClpFfiJsException{
```

Comment:
Since the structured one also throws the same error when a message cannot be decoded, let's document this `@throw` in the docstring at https://github.com/y-scope/clp-ffi-js/blob/d2556b2fb6d986182b247c660244dfbc923e37ea/src/clp_ffi_js/ir/StreamReader.hpp#L100-L101

Reply:
Sorry for the confusion. I mean we want to add this at https://github.com/y-scope/clp-ffi-js/blob/d2556b2fb6d986182b247c660244dfbc923e37ea/src/clp_ffi_js/ir/StreamReader.hpp#L99
```
@throw ClpFfiJsException if any error occurs.
```
or this may be more clear:
```
@throw ClpFfiJsException if a message cannot be decoded.
```